### DISCUSSION TITLE
how to create or update many-to-many relation?

### DISCUSSION THREAD

**Discussion by stardvst** [prisma/prisma#11420] [Q&A] [answered]
i have the following models, and many-to-many relation between them:
```prisma
model User {
  id            String    @id @default(cuid())
  name          String?
  email         String?   @unique
  followings    Artist[]
}

model Artist {
  id           String @id @default(cuid())
  name         String @unique
  spotifyId    String @unique
  followers    User[]
}
```
when a user logs into my app, i retrieve their current followed artists, and need to update my database.
i have managed to select artists data from database (for updating user <-> artist relation), sample data:
```ts
const followings = [
  {
    id: '...',
    name: 'MARINA',
    spotifyId: '6CwfuxIqcltXDGjfZsMd9A'
  },
  {
    id: '...',
    name: 'Dua Lipa',
    spotifyId: '6M2wZ9GZgrQXHCFfjv46we'
  },
]
```
this is my user object:
```ts
const user = {
  id: 'someId',
  name: 'someName',
  email: 'someEmail'
}
```
i tried to insert or update user <-> artist relation with this query but i'm getting `Bad Request` error:
```ts
await prisma.user.upsert({
	where: {
		email: user.email
	},
	create: {
		name: user.name,
		email: user.email,
		followings: {
			connectOrCreate: followings
		}
	},
	update: {
		followings: {
			connectOrCreate: followings
		}
	}
})
```
please advise what i need to do. thanks in advance.

**Comment by austincrim** [comment, marked as answer]
Hey @stardvst 👋 ,

Can you provide a specific error message? Is the error coming from Prisma or somewhere else?

Thanks!

**Reply by stardvst** [reply]
when i comment out the query, there's no error, so it should be coming from prisma.
this is the full stack trace that i get, but i don't think it's much helpful:
```
Error: Bad Request
    at saveFollowings (webpack-internal:///./lib/spotify.js:60:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async saveUserFollowings (webpack-internal:///./lib/spotify.js:100:9)
    at async eval (webpack-internal:///./pages/api/auth/[...nextauth].js
```
is there a prisma option to get more informative error message?

**Reply by austincrim** [reply]
You could enable [debugging output](https://www.prisma.io/docs/concepts/components/prisma-client/debugging#setting-the-debug-environment-variable) by prepending setting the `DEBUG=*` environment variable before running your app.

Could you provide the code snippet that's causing the error? Or a reproduction repo that I could look at?

**Reply by stardvst** [reply]
thanks for the env var.
after a while i found the culprit. in the example from docs `connectOrCreate` has `where` property: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#connect-or-create-a-record
the query works after adding it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
